.travis.yml

```diff
@@ -16,13 +28,26 @@ addons:
       - llvm-toolchain-precise
 
 before_install:
+  - if [[ $TRAVIS_OS_NAME == osx ]]; then TRAVIS_PYTHON_VERSION=3.5; fi
   - export TRAVIS_PYTHON_VERSION_MAJOR=${TRAVIS_PYTHON_VERSION%.[0-9]}
   - export TRAVIS_PYTHON_VERSION_MINOR=${TRAVIS_PYTHON_VERSION#[0-9].}
+
+  - >
+    if [[ $TRAVIS_OS_NAME == osx ]]; then
+      brew install python3
+      sudo ln -sf $(which python3) $(which python)
+      sudo ln -sf $(which pip3) $(which pip 2>/dev/null || echo /usr/bin/pip)
+    fi
+
   # https://github.com/coala/coala/issues/3183
   # Travis automatically installs the `requirements.txt` in "install" stage
   - cp requirements.txt requirements.orig
   - cat test-requirements.txt docs-requirements.txt >> requirements.txt
-  - sed -i '/^-r/d' requirements.txt
+  - sed '/^-r/d' requirements.txt > requirements.new && mv requirements.new requirements.txt
```

Comment:
It isnt supported by OSX.
I had a comment "sed -i isnt supported by OSX", but it seemed a bit silly because the line isnt using -i , and there are multiple cases of this, and anyone who changes it will get a broken build anyway.